### ISSUE TITLE
New definition of PrismaticJointR

### ISSUE THREAD

**Issue by vacary** [siconos/siconos#140]
This issue is related to the issue #120 

It seems that in the new definition of the PrismaticJointR, the axis of the joint is given with respect to the body--fixed frame of the first DS.

@radarsat1 could you confirm the definition ?

The examples in examples/Mechanics/JointsTests has to be updated.

**Comment by vacary** [comment]
start to fix a part of the examples : f57b5ac9caf9fe0f5494f376fcb45841c30a5edc

**Comment by radarsat1** [comment]
Ah, I'll check the examples.  But I'm surprised, my understanding was that the axis was always given in the frame of the first DS.  Looking back at old versions, e.g. `git show 4.0.0:./PrismaticJointR.cpp` gives,

~~~
  /** constructor from two dynamical systems and an axis
   *  \param d1 first  DynamicalSystem link by the  joint
   *  \param d2 second  DynamicalSystem link by the joint
   * \param axis SiconosVector of size 3 that defines the prismatic axis
   *  in the body  frame of d1 ?
   */
  PrismaticJointR(SP::NewtonEulerDS d1, SP::NewtonEulerDS d2, SP::SiconosVector axis);

~~~

Although I admit, that question mark was always a cause for concern.. ;)
I'll take a look.

**Comment by vacary** [comment]
Yes, the old definition of the PrismaticJointR was a little suspect. For all the other joints, we define the joints w.r.t to body--fixed frame of ds1. This was not the case for the Primactic joint. I think that the API is better right now, because it is uniform.

Thanks.

**Comment by radarsat1** [comment]
Regarding https://github.com/siconos/siconos/commit/be32ba992b4e943e11b78b9d594287adee421ae1#commitcomment-22160289

Sorry I should not have pushed a change to the reference so quickly.  It seems something has changed in the motion details even after correcting the gross motion behaviour to adjust the prismatic axis.  I will try ti figure out what, and whether the new or old reference should be kept.

**Comment by radarsat1** [comment]
Ok now I have verified that on dfbce0c082725023e38faad5813507d443db8ee8 the error for NE_3DS_3Knee_1Prism_MLCP_MoreauJeanCombinedProjection example with respect to the reference was 2.14828e-14, and that after 6fb8eef9c87dc03aa2e14abdb40408aefa617336 it changed to 1.4493e-08.

(To verify this, you must check out those commits and cherry-pick 2b1914685814c594c7ba1a28437c7c513a806350, and adjust the value of variable `axe1` -- without adjusting axe1, the error is 5.01185)

So perhaps it is a real change with regards to how the joint functions, and it was correct to update the reference file.

If I understood my own changes correctly, it should only have made a difference if the two DS did not have the same initial orientation. Indeed this seems to be the case in this example, as the beams start at different orientations.

**Comment by radarsat1** [comment]
I see that in ef2cc89d7a1c75dff506fc3b06575da028d36016 you (@vacary) have changed the reference file for PrismaticTest.cpp.  I was trying to fix that example yesterday but did not finish yet.  Is it correct now?

**Comment by vacary** [comment]
The example is not exactly doing the same computation since I didn't change the axis for the joint. The axis should be rotated with the body the give the same output. If you want to update, feel free

**Comment by radarsat1** [comment]
Ok, I'll try. Thanks.

**Comment by radarsat1** [comment]
Fixed in 24113a5c3366fb7f52e3ba2c5938eb39311fc2fa, but actually I would like to add a boolean to select absolute reference frame as in PivotJointR, so I'll leave this open until I do that.

**Comment by radarsat1** [comment]
Closed in 128310dab4.
